### PR TITLE
fix(DASer): Log sampling errors in DASer

### PR DESCRIPTION
**Pull request by walldiss**
Log errors in DASer instead of availability

### CODE REVIEW

**Review by Wondertan — DISMISSED** [review]

**Review by Wondertan — APPROVED** [review]

File: das/worker.go [hunk 1]
```diff
@@ -127,18 +128,25 @@ func (w *worker) sample(ctx context.Context, timeout time.Duration, height uint6
...
-				"failed to sample header",
-				"type", w.state.jobType,
-				"height", h.Height(),
-				"hash", h.Hash(),
-				"square width", len(h.DAH.RowRoots),
-				"data root", h.DAH.String(),
-				"err", err,
-				"finished (s)", time.Since(start),
-			)
+		if errors.Is(err, context.Canceled) {
```
Comment (cristaloleg) [inline das/worker.go:131]: Also `context.DeadlineExceeded` ?

**Review by cristaloleg — DISMISSED** [review]

File: das/worker.go (hunk 1 above)
Comment (renaynay) [reply das/worker.go:131]: Deadline exceeded would be a failure to retrieve share in given time so it would be a `share ErrNotAvailable`

**Review by renaynay — DISMISSED** [review]

File: das/worker.go [hunk 2]
```diff
@@ -127,18 +127,20 @@ func (w *worker) sample(ctx context.Context, timeout time.Duration, height uint6
...
+			return err
 		}
+
+		log.Errorw(
```
Comment (Wondertan) [inline das/worker.go:134]: Was about to comment that we need to see individual failures if we remove the logs from avail 👍🏻

File: share/availability/light/availability.go [hunk 3]
```diff
@@ -97,7 +97,6 @@ func (la *ShareAvailability) SharesAvailable(ctx context.Context, header *header
 	}
 
 	if err := dah.ValidateBasic(); err != nil {
-		log.Errorw("DAH validation failed", "error", err)
 		return err
 	}
```
Comment (Wondertan) [inline share/availability/light/availability.go:101]: Btw, we may want to remove this check entirely. Headers are guaranteed to be valid at this point.

Comment (walldiss) [reply share/availability/light/availability.go:101]: Removed in light simplification PR: https://github.com/celestiaorg/celestia-node/pull/3895

**Review by renaynay — APPROVED** [review]

**Review by Wondertan — APPROVED** [review]

**Pull request merged**